packages/block-library/src/image/editor.scss

```diff
@@ -103,3 +106,43 @@
 		margin: -$border-width;
 	}
 }
+
+// Although the float markup is different in the editor compared to the frontend,
+// this CSS uses the same technique to allow floats in a wide images context.
+// That is, the block retains its centering and max-width, and a child inside
+// is floated instead of the block itself.
+[data-type="core/image"][data-align="center"],
+[data-type="core/image"][data-align="left"],
+[data-type="core/image"][data-align="right"] {
```

Comment:
This feels like a specific fix to a generic problem, how is this solved for other floated blocks? Can't we have a generic solution to left/right aligned blocks?